src/models/EventsModel.ts

```diff
@@ -0,0 +1,78 @@
+import {cast, getSnapshot, toGenerator, types} from 'mobx-state-tree';
+import {EventModel} from './Events/EventModel';
+import {AsyncTask, runTask} from 'mst-async-task';
+import {EventsService, EventServiceTypes} from '../services';
+
+const EventsModel = types
+  .model('EventsModel')
+  .props({
+    allEvents: types.optional(types.array(EventModel), []),
+    getEventsTask: types.optional(AsyncTask, {}),
+    getEventDetailTask: types.optional(AsyncTask, {}),
+    getFeaturedEventTask: types.optional(AsyncTask, {}),
+    eventDetail: types.frozen(getSnapshot(EventModel.create({}))),
+    featuredEvent: types.frozen(getSnapshot(EventModel.create({}))),
+  })
+  .views(self => ({
+    get EventsCount() {
+      return self.allEvents.length;
+    },
+  }))
+  .actions(self => {
+    const getAllEvents = (
+      location?: string,
+      limit?: number,
+      isFree?: boolean,
+      showError?: boolean,
+      offset?: number,
+      sortOn?: EventServiceTypes.EventsSortOn,
+      sortBy?: EventServiceTypes.EventsSortBy,
+    ) =>
+      runTask(self.getEventsTask, function* () {
+        const response = yield* toGenerator(
+          EventsService.getAllEvents(
+            location,
+            limit,
+            isFree,
+            showError,
+            offset,
+            sortOn,
+            sortBy,
+          ),
+        );
+        if (response.stat === 'ok') {
+          self.allEvents = cast(response.data.events);
+        } else {
+          throw response;
```

Comment:
Why throw error here? If you wanted error to be thrown then we could have done that in api.ts itself. Lets discuss this